packages/sanity/src/core/studio/copyPaste/utils.ts

```diff
@@ -0,0 +1,121 @@
+import {type Path} from '@sanity/types'
+import {isString} from 'sanity'
+
+import {type CopyActionResult} from './types'
+
+const SANITY_CLIPBOARD_ITEM_TYPE = isWebKit()
+  ? 'text/plain'
+  : 'web application/sanity-studio-clipboard-item'
+
+export const getClipboardItem = async (): Promise<CopyActionResult | null> => {
+  const items = await window.navigator.clipboard.read()
+  const item = items.find((i) => i.types.includes(SANITY_CLIPBOARD_ITEM_TYPE))
+  const sanityItem = await item?.getType(SANITY_CLIPBOARD_ITEM_TYPE)
+  if (sanityItem) {
+    const text = await sanityItem.text()
+    return parseCopyResult(text) || null
+  }
+  return null
+}
```

Comment:
future product idea (outside of the scope of this PR): would be cool if this function could also accommodate pasting from other apps like google sheets. e.g. copying multiple lines from google sheets into an array field

cc @runeb